openai_adapter.py

```
@@ -0,0 +1,370 @@
+        for chunk in stream:
+            if chunk.choices and chunk.choices[0].delta.content:
+                yield chunk.choices[0].delta.content
+                
+        # Check if there are tool calls
+        # Note: This is simplified and may need enhancement for complex tool interactions
+
+class AdapterContext:
+    """Context wrapper that adapts between OpenAI and MCP contexts."""
+    def __init__(self, mcp_server, session_id=None):
+        self.mcp_server = mcp_server
+        self.session_id = session_id or generate_session_id()
```

Ruff (0.8.2)

Undefined name `generate_session_id`

(F821)